### ISSUE TITLE
Reset your Notification password email 

### ISSUE THREAD

**Issue by JuliannaR** [cds-snc/notification-api#259]
The link in the email message should appear as a link with title text that is understandable for the user.

<img width="820" alt="Screen Shot 2019-08-15 at 10 31 53 AM" src="https://user-images.githubusercontent.com/35537308/63103279-c67bd480-bf4a-11e9-9b72-79557cd23d60.png">

**Comment by maxneuvians** [comment]
The issue here is that these emails are the same templates that are used throughout the whole notification system. If you look here: https://notification.alpha.canada.ca/features/templates you can see it will take the raw link and `linkify` it automatically. This means that unless we change the template format to something like `http://...|title` or something like markdown `[Name](Link)' it will not be able make that change. 

I am all for making this change, but we will need to agree as a Team on the priority of this change as it affects a big part of how the system works.

**Comment by maxneuvians** [comment]
So it looks like you can use the existing markdown. For example

`Wot link? [Canada.ca](((LINK))) other test [Canada.ca](https://canada.ca)`

Will render out

![Screenshot from 2019-08-30 09-45-34](https://user-images.githubusercontent.com/867334/64025567-f3afb180-cb0a-11e9-98e6-f98421362457.png)

We just need to edit the templates.

**Comment by maxneuvians** [comment]
Can confirm that it does not remove the https or http as initially thought. 

So we just need to update all the templates with the right content:

ex: 

`[Password reset](((url)))`

Creates

![Screenshot from 2019-08-30 09-57-02](https://user-images.githubusercontent.com/867334/64026426-bf3cf500-cb0c-11e9-9c2b-5f9126852e14.png)

**Comment by maxneuvians** [comment]
The plain text version looks like this:

```
To reset your password, click this link:

Password reset: https://notification.alpha.canada.ca/new-password/
```

**Comment by maxneuvians** [comment]
To add a title as the issue requires follow this markdown style

`[LINK_TEXT](((url)) "LINK_TITLE")`

ex:

`[Password reset](((url)) "Password reset")`

**Comment by obrien-j** [comment]
Just adding that hiding the link behind text that looks like a link is somewhat spammy and might be an anti-pattern.

**Comment by willeybryan** [comment]
Lets hold off on this until we do some user testing around trust and some research into spam filers

**Comment by willeybryan** [comment]
Relevant Slack Conversation https://gcdigital.slack.com/archives/CAG3WKMC4/p1567783417100600

**Comment by willeybryan** [comment]
@AntoineGarciaSuarez @caycefischer 
If we haven't looked into this we should find out the best way to display links in emails so that they satisfy these criteria in order of importance: 
1. They are accessible to assistive technologies
2. Users trust them
3. They look good

**Comment by willeybryan** [comment]
As well all saw this was actually done
